### PR TITLE
feat: update BlockchainWriter DAO on blockDownloaded event

### LINKED ISSUE #776: Update BlockchainWriter entries with block information.
part of #707 

Update entries in the blockchainWriter collection where txId in `block.transactionIds` setting blockHash and blockHeight

### PR DESCRIPTION
**Pull request by krobi64**
[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

### Release
 Semantic release is enabled for this repository. Make sure you follow the right commit message convention. 
 We're using semantic-release's default — [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines).

## Description of Changes

Resolves #776 
References: #707 

Add an event listener for onBlockDownloaded in BlockchainWriter
Update the entries with the block height and block hash for each transactionId in the downloaded block.

### CODE REVIEW

File: src/BlockchainWriter/Controller.test.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/* tslint:disable:no-relative-imports */
+import { describe } from 'riteway'
+import { LightBlock } from '../Messaging/Messages'
```
Comment (kennylavender) [inline src/BlockchainWriter/Controller.test.ts]: add line break between node_module imports and relative imports

Comment (kennylavender) [inline src/BlockchainWriter/Controller.test.ts]: add linebreak between node module imports and relative imports
```suggestion

import { LightBlock } from '../Messaging/Messages'
```

File: src/BlockchainWriter/Controller.ts [hunk 2]
```diff
@@ -16,6 +17,13 @@ export interface ControllerConfiguration {
   readonly poetVersion: number
 }
 
+export const convertLightBlockToEntry = (lightBlock: LightBlock): Entry => (
```
Comment (lautarodragan) [inline src/BlockchainWriter/Controller.ts]: Can you place the parenthesis and curly braces in the same lines?

```
export const convertLightBlockToEntry = (lightBlock: LightBlock): Entry => ({
  blockHeight: lightBlock.height,
  blockHash: lightBlock.hash,
})
```

**Comment by kennylavender** [comment]
The error I am getting locally
```
(node:106) UnhandledPromiseRejectionWarning: MongoError: E11000 duplicate key error collection: poet.blockchainWriter index: ipfsDirectoryHash_1 dup key: { : null }
    at Function.create (/usr/src/app/node_modules/mongodb-core/lib/error.js:43:12)
    at toError (/usr/src/app/node_modules/mongodb/lib/utils.js:149:22)
    at coll.s.topology.update (/usr/src/app/node_modules/mongodb/lib/operations/collection_ops.js:1399:39)
    at /usr/src/app/node_modules/mongodb-core/lib/connection/pool.js:532:18
    at process._tickCallback (internal/process/next_tick.js:61:11)
```

File: src/BlockchainWriter/DAO.ts [hunk 3]
```diff
@@ -17,17 +19,27 @@ export class DAO {
...
       transactionCreationDate: null,
+      blockHash: null,
+      blockHeight: null,
     })
 
   readonly findTransactionlessEntry = () => this.blockchainWriterCollection.findOne({ txId: null })
 
+  readonly updateAllByTransactionId =  async (txIds: ReadonlyArray<string>, entry: Entry) => {
+    await this.blockchainWriterCollection.updateMany(
+      { txId: { $in: txIds} },
```
Comment (kennylavender) [inline src/BlockchainWriter/DAO.ts]:
```suggestion
      { txId: { $in: txIds } },
```

**Pull request merged**